packages/configuration-svc/src/infrastructure/mongodb_configset_repo.ts

```diff
@@ -0,0 +1,349 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+ Contributors
+ --------------
+ This is the official list (alphabetical ordering) of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Crosslake
+ - Pedro Sousa Barreto <pedrob@crosslaketech.com>
+
+ --------------
+ ******/
+
+ "use strict";
+
+import { ILogger } from "@mojaloop/logging-bc-public-types-lib";
+import { MongoClient, Collection } from "mongodb";
+import { IBoundedContextConfigSetRepository, IGlobalConfigSetRepository } from "@mojaloop/platform-configuration-bc-domain-lib";
+import { GlobalConfigurationSet, BoundedContextConfigurationSet } from "@mojaloop/platform-configuration-bc-public-types-lib";
+import semver from "semver";
+
+export class MongoConfigSetRepo implements IBoundedContextConfigSetRepository, IGlobalConfigSetRepository {
+    private _mongoUri: string;
+    private _logger: ILogger;
+    private _mongoClient: MongoClient;
+    protected _collectionGlobalConfigSets: Collection<GlobalConfigurationSet>;
+    protected _collectionBcConfigSets: Collection<BoundedContextConfigurationSet>;
+    private _initialized: boolean = false;
+    private readonly _databaseName: string = "configsets";
```

Comment:
We usually use the bounded context name for the main DB name, I suggest we change this to "platform-configuration".

Reply:
Thanks Pedro, we can later optimize this to use the DB directly.